types/express-serve-static-core/index.d.ts

```diff
@@ -97,17 +97,22 @@ type GetRouteParameter<S extends string> = RemoveTail<
 >;
 
 // prettier-ignore
-export type RouteParameters<Route extends string> = string extends Route ? ParamsDictionary
+export type RouteParameters<Route extends string> = Route extends `${infer Required}{${infer Optional}}${infer Next}` ?
+    ParseRouteParameters<Required> & Partial<ParseRouteParameters<Optional>> & RouteParameters<Next> :
+    ParseRouteParameters<Route>;
+
+export type ParseRouteParameters<Route extends string> = string extends Route ? ParamsDictionary
     : Route extends `${string}(${string}` ? ParamsDictionary // TODO: handling for regex parameters
     : Route extends `${string}:${infer Rest}` ?
             & (
                 GetRouteParameter<Rest> extends never ? ParamsDictionary
-                    : GetRouteParameter<Rest> extends `${infer ParamName}?` ? { [P in ParamName]?: string }
+                    : GetRouteParameter<Rest> extends `${infer ParamName}?` ? { [P in ParamName]?: string } // TODO: Remove old `?` handling when support for Express 4 is dropped
```

Comment:
Since `express` already has a v4 folder, it seems like this `?` case can be dropped in the v5 types?

Reply:
I could remove it, but then the optional types won't work for people that install latest (v4) of express and latest (v5) of types/express

Reply:
I'll update the comment at least

Reply:
I would really think people should not be installing two differing versions of express and its types; we explicitly kept the v4 types on DT. I understand it's an easy situation to get into, but, it doesn't make it correct...

Reply:
(But, leaving this comment is not _harmful_, it's just, to me, odd to still be talking about v4 in what are the v5 types)